messaging/models.py

```diff
@@ -74,16 +83,94 @@ class Meta:
     def __str__(self):
         return f"({self.pk}) {self.subject} ({self.sent_at or 'not sent'})"
 
-    def send(self):
-        self.recipient_count = (
-            Guardian.objects.filter(children__project=self.project).count()
-            if self.recipient_selection == Message.ALL
-            else random.randint(
-                0, 50
-            )  # TODO temporary thingy before actual send implementation
+    def send(self, *, force=False):
+        if self.sent_at and not force:
+            raise AlreadySentError()
+
+        guardians = self.get_recipient_guardians()
+
+        self.sent_at = timezone.now()
+        self.recipient_count = len(guardians)
+        self.save(update_fields=("sent_at", "recipient_count"))
+
+        for guardian in guardians:
+            with switch_language(self, guardian.language):
+                if guardian.language in getattr(
```

Comment:
I wonder would it be better to use the `send_notification`. How about creating a blank notification template with only `{{ body_text }}` inside?  I remember Venepaikka is using that one to handle custom message sending to end-user.

Reply:
Yeah good point. At first I was actually planning on doing that, but it requires the right kind of "dummy" template to exist in the db, which I don't like when it is not really needed. FWIW I think Ilmoitin should support this itself (sending a notification without a template).